src/state.py

```diff
@@ -11,6 +12,69 @@ def __init__(self):
     def add_student(self, student):
         self.students[student.student_id] = student
 
+    def get_score(self): 
+        score = 0
+        #if traded the subject as expected get score
+        for i in self.students.values(): 
+
+            #FLAGS 
+            alone = True        #the person is not with a friend
+            no_givin = True     #the person didn't givin any class 
+            no_target = True    #the person didn't get the target 
```

Comment:
The problems of restrictions were solved by adding some flags to the code

Reply:
From my point of view, these are flags for each student once it's inside the loop and changed there.